modules/cpmso-openstack-with-az-config.adoc

```diff
@@ -0,0 +1,69 @@
+//
+// * machine_management/control_plane_machine_management/cpmso-troubleshooting.adoc
+
+:_content-type: PROCEDURE
+[id="cpmso-openstack-with-az-config_{context}"]
+= Configuring {rh-openstack} with availability zones
+
+If you upgrade a cluster on {rh-openstack-first} from version 4.13 to 4.14 that uses root volumes with Availability Zones (AZs), you must manually configure machine plane machine resources to use control plane machine sets.
+
+If you are starting with the {product-title} cluster, version 4.14, this process is done automatically. If you are  upgrading and if the masters are deployed on multiple AZs, follow the steps.
+
+For the clusters deployed before the {product-title} version 4.14, the Installer correctly sets the same server group for all the masters, regardless of the number of AZs. Two of the three masters have an invalid `ServerGroup` in their Machine `ProviderSpec`.
+
+[NOTE]
+====
+Manually editing control plane machine resources does not update the control plane instances. However, it is necessary in order for the `cluster-control-plane-machine-set-operator` to correctly generate a `ControlPlaneMachineSet` for your cluster.
+====
+
+. Update the `ProviderSpec` for both `master-1` and `master-2` by setting the property `serverGroupName` of `spec.providerSpec` to the value of `master-0` `spec.providerSpec.serverGroupName`:
++
+[source,terminal]
+----
+$ oc edit machine/<cluster_id>-master-1 -n openshift-machine-api
+<make edits>
+$ oc edit machine/<cluster_id>-master-2 -n openshift-machine-api
+<make edits>
+----
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    apiVersion: machine.openshift.io/v1alpha1
+    availabilityZone: az0
+      cloudName: openstack
+    cloudsSecret:
+      name: openstack-cloud-credentials
+      namespace: openshift-machine-api
+    flavor: m1.xlarge
+    image: rhcos-4.14
+    kind: OpenstackProviderSpec
+    metadata:
+      creationTimestamp: null
+    networks:    
+    - filter: {}
+      subnets:  
+      - filter:
+          name: refarch-lv7q9-nodes
+          tags: openshiftClusterID=refarch-lv7q9
+    securityGroups:
+    - filter: {}
+      name: refarch-lv7q9-master
+    serverGroupName: refarch-lv7q9-master-az0 <1>
+    serverMetadata:
+      Name: refarch-lv7q9-master
+      openshiftClusterID: refarch-lv7q9
+    tags:
+    - openshiftClusterID=refarch-lv7q9
+    trunk: true
+    userDataSecret:
+      name: master-user-data
+----
+<1> Change the value.
+
+. In case you updated your control plane machine resources after install, follow these steps. 
+* In your {rh-openstack} cluster, find the AZ of the root volumes for your control plane machines and set it in the `availabilityZone` property of all the control plane machines.
+
+After aall the three Control plane Machine resources have the same correct ServerGroupName, your control plane is ready to be managed by the Cluster Control Plane Machine Set Operator. The Operator will generate a control plane machine set for you.
```

Comment:
This step should be replaced by the `oc describe` and `oc edit` part of the solution, I think--mark as active and then the Operator generates the set.

Reply:
Okay.